packages/perspective-viewer-d3fc/src/js/legend/styling/enableDragging.js

```diff
@@ -11,28 +11,61 @@ import * as d3 from "d3";
 import {isElementOverflowing} from "../../utils/utils";
 import {getChartElement} from "../../plugin/root";
 
+const margin = 10;
+const pinned = "pinned";
+const resizeForDraggingEvent = "resize.for-dragging";
+
 export function enableDragging(element) {
     const node = element.node();
     node.style.cursor = "move";
 
     const drag = d3.drag().on("drag", function() {
-        overrideDefaultResizingBehaviour(node);
         const [offsetX, offsetY] = enforceContainerBoundaries(this, d3.event.dx, d3.event.dy);
         this.style.left = `${this.offsetLeft + offsetX}px`;
         this.style.top = `${this.offsetTop + offsetY}px`;
+
+        const element = d3.select(this);
+        if (isNodeInTopRight(node)) {
+            pinNodeToTopRight(node, element);
+            return;
+        }
+
+        unpinNodeFromTopRight(node, element);
     });
 
     element.call(drag);
 }
 
-// Default behaviour for the legend is to remain pinned to the top right hand corner with a specific margin.
-// Once the legend has moved we cannot continue to use that css based approach.
-function overrideDefaultResizingBehaviour(node) {
-    d3.select(window).on("resize", function() {
-        const [offsetX, offsetY] = enforceContainerBoundaries(node, 0, 0);
-        node.style.left = `${node.offsetLeft + offsetX}px`;
-        node.style.top = `${node.offsetTop + offsetY}px`;
-    });
+function unpinNodeFromTopRight(node, element) {
+    if (element.attr(pinned) !== false) {
```

Comment:
I can see why you've done this, but it seems a shame to have to make up new attributes for DOM elements, when the legend itself is already a component that has its own state.

If instead of an exported function `enableDragging(element)`, you export a d3-style component `const draggable = draggableElement(); draggable(element);`, then the component can have state instead of having to attach state to the element.

```javascript
export function draggableComponent() {
  let pinned = false;
  const draggable = (element) => {
    // draggable code (may change the value of pinned)
  }
  return draggable;
}
```

Reply:
Would this then need to be saved in `settings`?